staging/src/k8s.io/api/extensions/v1beta1/types.go

```diff
@@ -1041,10 +1041,10 @@ type NetworkPolicy struct {
 	// +optional
 	Spec NetworkPolicySpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
 
-	// Status is the current state of the NetworkPolicy.
-	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
-	// +optional
-	Status NetworkPolicyStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+	// Status existed once on NetworkPolicy as an alpha feature, but got removed.
+	// This commented field should remain, so in a future if we decide to reimplement
+	// NetworkPolicyStatus a different protobuf name SHOULD be used!
+	// Status NetworkPolicyStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
"the future" not "a future"

Does the protobuf _name_ need to be different? Or just the ID?

Barring ourselves from being able to have a status field named "status" in the future is bad. Could very easily break tooling.

Actually, looking at all of the totally generic code that's getting removed... maybe it would be better to keep `.status` and only remove `.status.conditions`?

Reply:
hum I agree. When I was removing the code I was like "ok, it is kind of reasonable to keep an empty status". Not sure what others thinks on this, but I'm up to keep status and status rest and just remove Conditions from it.

Reply:
About protobuf name, I have no idea. I would need to check how is this persisted on etcd. Can take a look later.

Reply:
this is how others are doing

```
grep -ri tombstone staging/ | grep proto
staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:      // Fields is tombstoned to show why 5 is a reserved protobuf tag.
staging/src/k8s.io/api/core/v1/types.go:        // ReadOnly is tombstoned to show why 2 is a reserved protobuf tag.
staging/src/k8s.io/api/core/v1/types.go:        // TopologyKeys is tombstoned to show why 16 is reserved protobuf tag.
staging/src/k8s.io/api/core/v1/types.go:        // IPFamily is tombstoned to show why 15 is a reserved protobuf tag.
```

@liggitt may help us here

Reply:
So I think one thing to keep in mind here is that the constraint is not really "no one can ever use this field again", it's "no one can ever use this field again _in a different way_". But that's not really a problem: if anyone ever wanted to add back `.status.conditions` in the future, they would do it in exactly the same way we did it here...

So if we can get rid of `.status` and `.status.conditions`, with the constraint that if we bring back `.status.conditions` in the future, it has to look identical to the `.status.conditions` we have now, then that's fine.

Reply:
As I spoke privately with Dan, I'm still thinking (maybe too late) if we should abandon conditions on netpol. 

My rationale is that, while keeping .status or .status.conditions without anyone using it may lead to users confusion on "why nothing populates this field", removing it (or not having it) reduces the ability of external controllers to add some information here (let's say, some netpol tester project, or if some npp wants to implement something). 

I'm not sure if some api convention says that all objects should have status, but if they should, I think having conditions (even empty) the most generic approach. 

Honestly, I couldn't think enough if the value of keeping it, as an api convention is bigger than not having it right now and needing to maybe reimplement it in a future